### PR TITLE
Master migration of account asset to new api - dve

### PR DESCRIPTION
**Pull request by Icallhimtest**
Migration to new api of module account_asset.

Fixed views :
- on account.asset.asset form view, field method_time, in "Depreciation Information"
  page: use radio widget instead (without it, you couldn't change the field)
- on account.asset.category form view: prorata field hidden when required, and set
  to false when method_time is on "Ending Date". (It was previously possible for
  the prorata to be true while the method_time was "Ending Date", which is invalid)

Task : https://www.odoo.com/web#id=27339&view_type=form&model=project.task&action=333&active_id=248&menu_id=4720
## 

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account_asset/__openerp__.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
     'category': 'Accounting',
     'sequence': 32,
     'demo': [
-        'account_asset_demo.yml',
+        'test/account_asset_demo.yml',
```
Comment (tde-banana-odoo) [inline addons/account_asset/__openerp__.py]: If it is a demo file, it should be in the data directory. Even if it is used in tests.

File: addons/account_asset/__openerp__.py [hunk 2]
```diff
@@ -24,13 +24,13 @@
...
-        'views/account_asset_invoice_view.xml',
-        'report/account_asset_report_view.xml',
-        'views/account_asset.xml',
-        'account_asset_cron.xml',
+        'wizard/account_asset_change_duration_views.xml',
+        'wizard/wizard_asset_compute_views.xml',
+        'views/account_asset_views.xml',
+        'views/account_asset_invoice_views.xml',
+        'report/account_asset_report_views.xml',
+        'views/account_asset_templates.xml',
```
Comment (tde-banana-odoo) [inline addons/account_asset/__openerp__.py]: Just to be maniacal: put report views after all views :) .

File: addons/account_asset/data/account_asset_cron.xml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding='UTF-8'?>
 <odoo>
-	<record model="ir.cron" id="account_asset_cron">
+	<record id="account_asset_cron" model="ir.cron">
```
Comment (tde-banana-odoo) [inline addons/account_asset/data/account_asset_cron.xml]: Unnecessary diff.

File: addons/account_asset/models/account_asset.py [hunk 4]
```diff
@@ -15,9 +16,9 @@ class AccountAssetCategory(models.Model):
...
-    account_asset_id = fields.Many2one('account.account', string='Asset Account', required=True, domain=[('internal_type','=','other'), ('deprecated', '=', False)], help="Account used to record the purchase of the asset at its original price.")
-    account_depreciation_id = fields.Many2one('account.account', string='Depreciation Entries: Asset Account', required=True, domain=[('internal_type','=','other'), ('deprecated', '=', False)], help="Account used in the depreciation entries, to decrease the asset value.")
-    account_depreciation_expense_id = fields.Many2one('account.account', string='Depreciation Entries: Expense Account', required=True, domain=[('internal_type','=','other'), ('deprecated', '=', False)], oldname='account_income_recognition_id', help="Account used in the periodical entries, to record a part of the asset as expense.")
+    account_asset_id = fields.Many2one('account.account', string='Asset Account', required=True, domain=[('internal_type', '=', 'other'), ('deprecated', '=', False)], help="Account used to record the purchase of the asset at its original price.")
```
Comment (tde-banana-odoo) [inline addons/account_asset/models/account_asset.py]: While we are at it, we could use a multi line definition. We do not limitate ourselves to 80 characters, but trying to stick on 100 or 120 could be interesting.

File: addons/account_asset/models/account_asset.py [hunk 5]
```diff
@@ -38,11 +39,11 @@ class AccountAssetCategory(models.Model):
     type = fields.Selection([('sale', 'Sale: Revenue Recognition'), ('purchase', 'Purchase: Asset')], required=True, index=True, default='purchase')
 
     @api.onchange('account_asset_id')
-    def onchange_account_asset(self):
+    def _onchange_account_asset(self):
```
Comment (tde-banana-odoo) [inline addons/account_asset/models/account_asset.py]: I would avoid renaming existing code already in new API.

File: addons/account_asset/models/account_asset.py [hunk 6]
```diff
@@ -38,11 +39,11 @@ class AccountAssetCategory(models.Model):
...
         self.account_depreciation_id = self.account_asset_id
 
     @api.onchange('type')
-    def onchange_type(self):
+    def _onchange_type(self):
```
Comment (tde-banana-odoo) [inline addons/account_asset/models/account_asset.py]: Same here, and apply to other changes of this kind.

File: addons/account_asset/models/account_asset.py [hunk 7]
```diff
@@ -55,7 +56,7 @@ class AccountAssetAsset(models.Model):
     _description = 'Asset/Revenue Recognition'
     _inherit = ['mail.thread', 'ir.needaction_mixin']
 
-    entry_count = fields.Integer(compute='_entry_count', string='# Asset Entries')
+    entry_count = fields.Integer(compute='_compute_entry_count', string='# Asset Entries')
```
Comment (tde-banana-odoo) [inline addons/account_asset/models/account_asset.py]: Same here.

File: addons/account_asset/models/account_asset_invoice.py [hunk 8]
```diff
@@ -1,12 +1,11 @@
...
 
 from datetime import datetime
 from dateutil.relativedelta import relativedelta
-from openerp import api, fields, models
-from openerp.exceptions import Warning
-from openerp.tools.translate import _
-import openerp.addons.decimal_precision as dp
-from openerp.tools import DEFAULT_SERVER_DATE_FORMAT as DF
+from odoo import api, fields, models
+import odoo.addons.decimal_precision as dp
```
Comment (tde-banana-odoo) [inline addons/account_asset/models/account_asset_invoice.py]: Maybe separate from and direct import statements.

File: addons/account_asset/models/account_asset_invoice.py [hunk 9]
```diff
@@ -49,43 +48,22 @@ def _get_asset_date(self):
...
-
-    @api.onchange('asset_category_id')
-    def onchange_asset_category_id(self):
-        if not self.asset_category_id:
-            self.account_id = self.get_invoice_line_account(self.invoice_id.type, self.product_id, self.invoice_id.fiscal_position_id, self.invoice_id.company_id)
-        if self.invoice_id.type == 'out_invoice' and self.asset_category_id:
-            self.account_id = self.asset_category_id.account_asset_id.id
-        elif self.invoice_id.type == 'in_invoice' and self.asset_category_id:
-            self.account_id = self.asset_category_id.account_depreciation_id.id
+    # this method gets overrided later ... -> no point, to remove
```
Comment (tde-banana-odoo) [inline addons/account_asset/models/account_asset_invoice.py]: Whut ?

File: addons/account_asset/report/account_asset_report_views.xml [hunk 10]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
 
-    <record model="ir.ui.view" id="action_account_asset_report_pivot">
+    <record id="action_account_asset_report_pivot" model="ir.ui.view">
```
Comment (tde-banana-odoo) [inline addons/account_asset/report/account_asset_report_views.xml]: Unnecessary diff, and other also.

File: addons/account_asset/models/account_asset.py [hunk 11]
```diff
@@ -102,38 +102,27 @@ def _compute_entry_count(self):
 
     @api.depends('value', 'salvage_value', 'depreciation_line_ids.move_check', 'depreciation_line_ids.amount')
     def _compute_amount_residual(self):
-        total_amount = 0.0
-        for line in self.depreciation_line_ids:
-            if line.move_check:
-                total_amount += line.amount
-        self.value_residual = self.value - total_amount - self.salvage_value
+        for asset in self:
```
Comment (tde-banana-odoo) [inline addons/account_asset/models/account_asset.py]: If you do some refactoring ... total_amount = sum(asset.depreciation_line_ids.filtered(lambda line: line.move_check).mapped('amount') ?

File: addons/account_asset/models/account_asset_invoice.py [hunk 12]
```diff
@@ -79,26 +68,24 @@ def _onchange_asset_category_id(self):
...
-                'date': self.asset_start_date or self.invoice_id.date_invoice,
-                'invoice_id': self.invoice_id.id,
-            }
-            changed_vals = self.env['account.asset.asset']._onchange_category_id_values(vals['category_id'])
-            vals.update(changed_vals['value'])
-            asset = self.env['account.asset.asset'].create(vals)
-            if self.asset_category_id.open_asset:
-                asset.validate()
-        return True
+        for line in self:
```
Comment (tde-banana-odoo) [inline addons/account_asset/models/account_asset_invoice.py]: Actually you change the return type. Maybe someone in the world waits for a "True" result ... pay attention to cleaning, it should not change results and stuff like that.

File: addons/account_asset/report/account_asset_report.py [hunk 13]
```diff
@@ -26,9 +26,9 @@ class AssetAssetReport(models.Model):
     unposted_value = fields.Float(string='Unposted Amount', readonly=True)
     company_id = fields.Many2one('res.company', string='Company', readonly=True)
 
-    def init(self, cr):
-        tools.drop_view_if_exists(cr, 'asset_asset_report') # why is this called if the instruction 2 lines below is create or REPLACE view ?
-        cr.execute("""
+    @api.model
```
Comment (tde-banana-odoo) [inline addons/account_asset/report/account_asset_report.py]: This seems incorrect, old definition was an api.model_cr. There was no context, was it missing somewhere ?

File: addons/account_asset/wizard/account_asset_change_duration.py [hunk 14]
```diff
@@ -15,13 +15,13 @@ class AssetModify(models.TransientModel):
...
     method_end = fields.Date(string='Ending date')
-    asset_method_time = fields.Char(compute='_get_asset_method_time', string='Asset Method Time', readonly=True)
+    asset_method_time = fields.Char(compute='_compute_asset_method_time', string='Asset Method Time', readonly=True)
 
-    @api.one
-    def _get_asset_method_time(self):
-        if self.env.context.get('active_id'):
-            asset = self.env['account.asset.asset'].browse(self.env.context.get('active_id'))
-            self.asset_method_time = asset.method_time
+    def _compute_asset_method_time(self):
```
Comment (tde-banana-odoo) [inline addons/account_asset/wizard/account_asset_change_duration.py]: What is the purpose of that change ?

File: addons/account_asset/wizard/account_asset_change_duration.py [hunk 15]
```diff
@@ -56,7 +56,7 @@ def default_get(self, fields):
         if 'method_end' in fields and asset.method_time == 'end':
             res.update({'method_end': asset.method_end})
         if self.env.context.get('active_id'):
-            res['asset_method_time'] = self._get_asset_method_time()
+            res['asset_method_time'] = self._compute_asset_method_time()
```
Comment (tde-banana-odoo) [inline addons/account_asset/wizard/account_asset_change_duration.py]: We set None as a value in the dict ? What is its use ? Shouldn't it be res[..] = self.asset_method_time ?

File: addons/account_asset/models/account_asset.py (hunk 4 above)
Comment (Icallhimtest) [inline addons/account_asset/models/account_asset.py]: Agreed, this could be interesting to add to the guidelines.
